### DISCUSSION TITLE
Error on vercel

### DISCUSSION THREAD

**Discussion by Arshan-K** [vercel/community#4331] [Help] [answered]
### Summary

When i run my node.js code in my laptop locally using localhost......It works fine but When i uploaded my code on github then on Vercel to host it shows 404: NOT_FOUND
Code: NOT_FOUND
I tried various methods to fix it but no use. There is database connection and all but it is working can anyone help me please ?
![Screenshot (90)](https://github.com/vercel/community/assets/126375801/7a231397-a3b7-44cb-b67b-1c637636a6e5)

### Example

https://certificate-veri.vercel.app/

### Steps to Reproduce

const express = require('express');
const app = express();
const fs = require('fs');
const path = require('path'); 
const mysql = require('mysql2');
const cors = require('cors');

require("dotenv").config();
app.use(express.static(path.join(__dirname, 'public')));

const connection = mysql.createConnection({
    host: process.env.DB_HOST,
    user: process.env.DB_USER,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_DATABASE,
    port: process.env.DB_PORT,
    waitForConnections: true
})

  connection.connect((err) => {
    if (err) {
      console.error("Error connecting to the database:", err);
    } else {
      console.log("Connected to the database!");
    }
  });
  
  app.use(cors());
app.use(express.static(path.join(__dirname, 'public'))); // Serve static files from the 'public' folder

app.get('/favicon.ico', (req, res) => {
    res.status(204).end(); 
});

app.get('/getData', (req, res) => {
    const Reference_Number = req.query.rid;
    
        if (Reference_Number) {

            const sql = 'SELECT * FROM data WHERE `COL 1` = ?';
            connection.query(sql, [Reference_Number], (err, results) => {
            
            if (err) {
                console.error(err);
                res.status(500).send('Internal Server Error');       //data fecthing from database 
                return;
            }

           
            if (results.length === 0) {
                res.status(404).send('Data not !!!! found');
                return;
            }

            const data = results[0]; // Assuming there's only one matching row
            const result = {
                Certificate_Number: data.Certificate_Number,
                Name: data.Name,
                Designation: data.Designation,
                Institute: data.Institute,
                Workshop_Organiser: data.Workshop_Organiser,
                Workshop_Name: data.Workshop_Name,
                Workshop_Date: data.Workshop_Date,
            };

            res.json(results);
        });
    } else {
        res.status(400).send('Bad Request');
    }
});

const myHTTP_PORT = process.env.HTTP_PORT  
const MYSQL_PORT = process.env.DB_PORT  // Use port 3306 as a default for MySQL

app.listen(myHTTP_PORT, () => {
    console.log(`HTTP Server is running on port ${myHTTP_PORT}`);
});
![Screenshot (92)](https://github.com/vercel/community/assets/126375801/e9a828dc-803f-41b6-8c87-5bca6a60f43a)
![Screenshot (93)](https://github.com/vercel/community/assets/126375801/b4ebe0bc-50e7-43a5-93ea-e427becee4a2)

**Comment by amyegan** [comment, marked as answer]
When you run the project locally, do you use [`vercel dev`](https://vercel.com/docs/cli/dev) or a different command?

Express needs some special configuration to work in a serverless environment instead of a more traditional server. That's a common cause of deployment errors with Express apps. We have a guide that walks through the recommended setup: https://vercel.com/guides/using-express-with-vercel

**Comment by Arshan-K** [comment]
I directly run locally using npm start......since I had given start
instructions in script.
So I don't use anything else.

**Reply by amyegan** [reply]
Try installing [Vercel CLI](https://vercel.com/docs/cli) and running the project with `vercel dev --debug`. That will run it on localhost and log additional details that can give us more clues about what is going wrong.

**Comment by Arshan-K** [comment]
Thanks, I'll check it out.

**Comment by javierfuentesm** [comment]
@amyegan Hi, im getting a similar error I have an API in Vercel and it was working fine but since Friday it has started to fail. it randomly throws 404. but when I open that link I get the response, I don't even get a timeout, so it is confusing because it works locally, it works on production but when I try to use the API from my discord bot I get a 404 ... 
![image](https://github.com/vercel/community/assets/19244527/cfcb44ac-ead2-48b2-a9e6-524d21caf2ba)

**Reply by amyegan** [reply]
Hi @javierfuentesm. Would you mind starting a [new discussion](https://github.com/vercel/community/discussions/new) with more info about your project? A [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) would be very helpful for debugging if you can share one. Since your deployed version seems to work in some situations but not others, I think this problem is not quite the same as the original one here.

**Reply by javierfuentesm** [reply]
I'm sorry, I found the error and I feel so stupid 🫠🫠, when I was opening the link in my browser in Opera it was returning info but it was the cache, after I opened the page in incognito mode I was getting the same as in vercel

**Reply by amyegan** [reply]
Ah! I'm glad it was a fairly simple problem and you already found a solution 😄
